### ISSUE TITLE
[BUG]: Model.create reports error for a field of type Date

### ISSUE THREAD

**Issue by matthelliwell2** [sensedeep/dynamodb-onetable#84]
**Describe the bug**
I have a schema containing a model with a field of type Date. When I use the create method, the compiler complains that "Type 'Date' is not assignable to type 'undefined'."

**To Reproduce**
The code below doesn't compile and reports the following error:

src/repo/Error Schema.ts:21:59 - error TS2322: Type 'Date' is not assignable to type 'undefined'.

21 void Foo.create({ pk: 'pk', sk: 'sk', strField: 'field1', dateField: new Date() })
                                                             ~~~~~~~~~

  src/repo/Error Schema.ts:11:13
    11             dateField: { type: Date, required: true }
                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    The expected type comes from property 'dateField' which is declared here on type 'Entity<{ pk: { type: StringConstructor; required: boolean; }; sk: { type: StringConstructor; required: boolean; }; strField: { type: StringConstructor; required: boolean; }; dateField: { ...; }; }>'

```
import { Entity, Model, Table } from 'dynamodb-onetable'

const mySchema = {
    indexes: { primary: { hash: 'pk', sort: 'sk' } },
    models: {
        Model1: {
            pk: { type: String, required: true },
            sk: { type: String, required: true },

            strField: { type: String, required: true },
            dateField: { type: Date, required: true }
        }
    }
}

type MyType = Entity<typeof mySchema.models.Model1>

export const table = new Table({ name: 'MyTable', delimiter: '#', timestamps: true, isoDates: true, schema: mySchema })
export const Foo: Model<MyType> = table.getModel<MyType>('Foo')

void Foo.create({ pk: 'pk', sk: 'sk', strField: 'field1', dateField: new Date() })
```

**Expected behavior**
Code to compile without errors

**Environment (please complete the following information):**
 - OS - windows
 - Node Version - 14
 - Typescript - 4.3.5

**Comment by mobsense** [comment]
Thanks for reporting. Fix coming.

Also, your line below

export const Foo: Model<MyType> = table.getModel<MyType>('Foo')

should be:

export const Foo: Model<MyType> = table.getModel<MyType>('Model1')

**Comment by mobsense** [comment]
Fixed in 1.6.2

**Comment by matthelliwell2** [comment]
Thanks for the quick fix, my code now compiles without a problem.
